src/core/connection.c

```diff
@@ -1612,7 +1612,7 @@ QuicConnTryClose(
             QuicConnLogStatistics(Connection);
         }
 
-        if (Flags & QUIC_CLOSE_APPLICATION) {
+        if (Flags & QUIC_CLOSE_APPLICATION) { // NOLINT
```

Comment:
Why is it flagging so many if statements with // NOLINT? This flags one _maybe_ I can understand, but the one below on line 2036 is odd to me.

Reply:
Because `QUIC_CLOSE_APPLICATION` is a (signed) `int`, and bitops on signed values are undefined.

Reply:
The one on 2036 (and many others) is because both paths of the `if` have the same code - `QuicTraceEvent` is a no-op on Linux.

I can comment out this check instead?

Reply:
The logging logic here is important. What's the correct way to fix this logic? Why is it treating `QUIC_CLOSE_APPLICATION` as a signed int?

Reply:
The issue here is that both paths of the `if` are identical, because `QuicTraceEvent` is a no-op. That is what the `NOLINT` suppresses.

(I was wrong on the bitops thing; must have looked at some other code piece.)